### DISCUSSION TITLE
Query on theia customization

### DISCUSSION THREAD

**Discussion by badrinarayan-R83** [eclipse-theia/theia#13790] [General] [answered]
Right now, when I launch theia browser example, the IDE occupies entire browser space.
Is there any way we could customize the IDE, so that the entire theia IDE is positioned inside some frame at the bottom of the browser. We may want to add other elements inside the top frame of the browser

**Comment by msujew** [comment]
Hey @badrinarayan-R83,

Theia is not designed to run as part of another frontend app, there are a bunch of technical and architectural design reasons for that.

However, something that works quite well (and a lot of adopters do) is embed Theia via an `iframe`. You just need to obtain the src URL of the workspace and then you should be good to go. Afterwards, you can use the iframe/browser messaging API in case you need to communicate with the frontend of your Theia App from the "outer" application.

**Reply by badrinarayan-R83** [reply]
Thanks for the quick reply. How do we embed theia inside an iframe ? Does it required some change in any theia related file ?  
I mean can we change something in theia related file so that when we launch http://localhost:3000, theia IDE appears inside an iframe at the position where we want

**Reply by msujew** [reply, marked as answer]
You would have a surrounding application that then points an iframe to `localhost:3000`. It's nothing you do inside of Theia. I don't really know any other way to explain that.
